client/src/contexts/user/UserProvider.js

```diff
@@ -0,0 +1,28 @@
+import React, { useState } from 'react';
+import UserContext from './UserContext';
+import PropTypes from 'prop-types';
+
+function UserProvider({ children }) {
+  const [user, setUser] = useState(null);
+
+  function saveUser(user) {
```

Comment:
You could directly use `setUser` instead of creating `saveUser`. These functions are equal.

Reply:
If I try to use `setState(user)` instead of `function saveUser(user){}` I get an error with too many re-renders... 😞

Reply:
just return `setUser` instead of `saveUser`